miscellaneous/build_templates/analysis.yml

```diff
@@ -8,6 +8,7 @@ jobs:
     pool:
       name: 00-OSIManaged-Test
       demands: COMPUTERNAME -equals $(buildAgent)
+    dependsOn: Tests
```

Comment:
Is this necessary? Keep in mind changes in this template will affect all our builds, so if a pipeline didn't have a stage named "Tests" this would break it.

Reply:
The tests required that the utility run before them, so we needed this to enforce the order of the pipeline.

Reply:
I don't think I understand; the pipeline described in this file doesn't run your tests or your utility.

Reply:
We noticed that when running the build that it could run the tests before the utility because they are in separate jobs. We decided to add a dependsOn in the Tests job so that it depends on the utility, and then we saw the analysis run first, so we added dependsOn here as well. This job template is referenced in the the pipeline, and it looks like in all the samples' pipelines. The other option is to merge the 2 jobs so that we are back to having only 2 jobs, Tests and the analysis, which I think should allow us to remove the dependsOn you see here. I think the proper way to handle multiple jobs is to explicitly specify the order using dependsOn, so leaving this as is should be ok.

Reply:
Does it actually fail if the analysis runs before the utility? It shouldn't be a problem, since the analysis shouldn't require anything from the utility. The analysis template itself is designed to do everything necessary for analysis, because some pipelines _only_ run the analysis template on the internal build agent and other build jobs are run on other machines. If we need to run two separate builds for this sample in order for analysis to see everything, we should probably adjust the analysis template in a different way. I definitely don't want the analysis template to depend on anything. We can discuss in detail via chat if you'd like to determine a solution.

Reply:
We can remove the changes here and test. It sounds like we may have had a misunderstanding while running the pipeline.